lib/spack/spack/spec.py

```diff
@@ -1666,6 +1656,9 @@ def _add_flag(self, name, value, propagate):
             self._set_architecture(os=value)
         elif name == "target":
             self._set_architecture(target=value)
+        elif name == "namespace":
+            self.namespace = value
+            assert not propagate
```

Comment:
This assertion is redundant to the error raised 15 lines earlier.

Reply:
oops did all the other changes and forgot that.  fixed!